lib/dry/transaction/step.rb

```diff
@@ -13,24 +13,26 @@ class Step
       attr_reader :operation_name
       attr_reader :operation
       attr_reader :options
+      attr_reader :block
       attr_reader :call_args
 
-      def initialize(step_adapter, step_name, operation_name, operation, options, call_args = [])
+      def initialize(step_adapter, step_name, operation_name, operation, options, call_args = [], &block)
         @step_adapter = step_adapter
         @step_name = step_name
         @operation_name = operation_name
         @operation = operation
         @options = options
+        @block = block
         @call_args = call_args
       end
 
-      def with_call_args(*call_args)
-        self.class.new(step_adapter, step_name, operation_name, operation, options, call_args)
+      def with_call_args(*call_args, &block)
```

Comment:
Was there a reason you've added block to this method's arguments, @am-kantox? As far as I can tell, there should be no need for `&block` to be an argument, Since there's no way to pass a block as an extra step argument and `block` in the line below is a reference to the `@block` we've already captured in the initializer.